funfact/model/_factorization.py

```diff
@@ -133,6 +135,21 @@ def ndim(self):
         '''The dimensionality of the result tensor.'''
         return self.tsrex.ndim
 
+    def penalty(self, sum_leafs: bool = True, sum_vec=None):
```

Comment:
Similar to the comment above, could we simply invoke the conditions on the `data` tensor while iterating over `dfs_filter`?

Reply:
Done.